File: VB/GridControlWithBar/PropertyGridControlDescendant/MultiRecordViewInfoDescendant.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraVerticalGrid
Imports DevExpress.XtraVerticalGrid.ViewInfo
Imports System.Drawing

Namespace GridControlWithBar
	Public Class MultiRecordViewInfoDescendant
		Inherits MultiRecordViewInfo
		Public Sub New(ByVal grid As VGridControlBase, ByVal isPrinting As Boolean)
			MyBase.New(grid, isPrinting)

		End Sub

		Protected Overrides Sub CreateBand(ByVal bandHeight As Integer, ByVal bandWidth As Integer)
			Dim left As Integer = (If(ViewRects.BandRects.Count = 0, ViewRects.Client.Left, (CType(ViewRects.BandRects(ViewRects.BandRects.Count - 1), Rectangle)).Right))
			Dim br As New Rectangle(left, ViewRects.Client.Top + (TryCast(Grid, PropertyGridControlDescendant)).BarHeight, bandWidth, Math.Min(bandHeight, ViewRects.Client.Height))
			ViewRects.BandRects.Add(br)

		End Sub
	End Class
End Namespace

File: CS/GridControlWithBar/PropertyGridControlDescendant/MultiRecordViewInfoDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraVerticalGrid;
using DevExpress.XtraVerticalGrid.ViewInfo;
using System.Drawing;

namespace GridControlWithBar
{
    public class MultiRecordViewInfoDescendant: MultiRecordViewInfo
    {
        public MultiRecordViewInfoDescendant(VGridControlBase grid, bool isPrinting)
            : base(grid, isPrinting)
        {
            
        }

        protected override void CreateBand(int bandHeight, int bandWidth)
        {
            int left = (ViewRects.BandRects.Count == 0 ? ViewRects.Client.Left :
          ((Rectangle)ViewRects.BandRects[ViewRects.BandRects.Count - 1]).Right);
            Rectangle br = new Rectangle(left, ViewRects.Client.Top + (Grid as PropertyGridControlDescendant).BarHeight, bandWidth, Math.Min(bandHeight, ViewRects.Client.Height));
            ViewRects.BandRects.Add(br);

        }
    }
}
